src/main/java/com/ericsson/gerrit/plugins/eiffel/storage/EventStorage.java

```diff
@@ -0,0 +1,98 @@
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.NoSuchElementException;
+import com.ericsson.gerrit.plugins.eiffel.handlers.DatabaseHandler;
+import com.ericsson.gerrit.plugins.eiffel.handlers.Table;
+
+public abstract class EventStorage {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(EventStorage.class);
+    protected File pluginDir;
+
+    public EventStorage(File pluginDir) {
+        this.pluginDir = pluginDir;
+    }
+
+    public abstract String getEventId(String project, String searchCriteria) throws NoSuchElementException, ConnectException, FileNotFoundException;
+
+    public abstract void saveEventId(String eiffelEventId, EiffelEvent eiffelEvent)
+            throws NoSuchElementException, SQLException, ConnectException;
+
+    protected String getLastSavedEiffelEvent(String project, String searchCriteria,
+            Table tableName)
+            throws NoSuchElementException, FileNotFoundException, ConnectException {
+        try {
+            return getEventId(project, searchCriteria, tableName);
+        } catch (NoSuchElementException e) {
+            throw new NoSuchElementException(
+                    "Database did not return any value for this query\n" + "Exception Message:" + e.getMessage());
+        }
+    }
+
+    protected void findAndUpdateEiffelEvent(String project, String searchCriteria,
+            String eiffelEvent,
+            Table tableName) throws NoSuchElementException, ConnectException, SQLException {
+        DatabaseHandler dBHandler = new DatabaseHandler(pluginDir, project);
+
+        saveEventToDatabase(project, searchCriteria, eiffelEvent, tableName, dBHandler);
+    }
+
+    private void saveEventToDatabase(String project, String searchCriteria, String eiffelEvent,
+            Table tableName, DatabaseHandler dBHandler) throws ConnectException, SQLException {
+        String oldEvent = getOldEventId(dBHandler, tableName, searchCriteria);
+
+        if (!oldEvent.isEmpty()) {
+            dBHandler.updateInto(tableName, searchCriteria, eiffelEvent);
+            LOGGER.info("Replaced old event id '{}' with new event if '{}', for project '{}', and branch '{}'.",
+                    oldEvent, eiffelEvent, project, searchCriteria);
+        } else {
+            dBHandler.insertInto(tableName, searchCriteria, eiffelEvent);
+            LOGGER.info("Saved eiffel event with id '{}', for project '{}', and branch '{}'.", eiffelEvent, project,
+                    searchCriteria);
+        }
+    }
+
+    private String getOldEventId(DatabaseHandler dBHandler, Table tableName, String searchCriteria) throws ConnectException {
+        try {
+            String oldEvent = dBHandler.getEventID(tableName, searchCriteria);
+            return oldEvent;
+        } catch (NoSuchElementException e) {
+            LOGGER.debug("No previous event was saved, creating a new entry");
```

Comment:
I would like to rephrase this error message to "No old event found" or something. No new entry is created in this method.

Reply:
"No old event found"